### DISCUSSION TITLE
12 of my Best Suggestions (previously 7 Best Suggestions)

### DISCUSSION THREAD

**Discussion by Krzeszny** [FakeFishGames/Barotrauma#5157] [Ideas and suggestions]
1. Adding a switch/manual fuse in the repair UI that **isolates/turns off junction boxes** (and other machines) to prevent electrocution. This is balanced by the need to cut off the power while repairing. It also makes the repair UI more interactive.

2. Picking up disconnected ends of wires so that there's no need to disconnect both ends to **reconnect a wire's end** to something else.

3. ~~A **handheld terminal** for probing wire signals, used by hovering the cursor over a wire's end. Using a stationary terminal is very inconvenient.~~ Implemented.

4. An animation for the **Lever** that shows its state, maybe with a light like the Switch (either a green power light, or a red _power off_ light, or green/red switching lights.)

5. ~~A command that tells AI crew to **prepare diving suits**. When activated a second time it tells them to put the diving suits away.~~ Implemented.

6. ~~A setting in the options menu to choose the % at which the **Oxygen Low alert** starts beeping in a diving suit. It's very annoying at 5% because non-empty tanks don't stack and we have to listen to the beeping between 5% and 0%. Mod support would be mandatory (one solution would be to choose one setting for all equipment tagged 'diving' or at least 'deepdiving' because masks don't beep.~~ Partially implemented.

7. Wrecks of vanilla subs to add **wreck variation**.

8. ~~A parameter for the **navigation terminal** that adds an optional mineral sonar (either the one from the sonar terminal or a simplified one) and here's why:~~

~~Having to use the handheld sonar to detect minerals from the sub has many **flaws**:
a) the handheld sonar can currently detect minerals outside of the sub and using it like that is cheesy to even cheaty
b) a handheld tool can detect what the main sonar can't? Sure, the stand-alone sonar terminal can detect minerals but it's neither obvious nor viable on subs that don't have a sonar terminal by default
c) players have to keep the handheld sonar on active or ping it every few seconds, both being annoying (unless the sonar terminal is installed and has an operator or is next to the navigation terminal, the latter being annoying because of the double ping)
d) **no merits** to not having the option to add it to the nav terminal (yes, it's a joke that one of the flaws is the lack of merits)
e) sonars are normally deafeningly-loud (using a handheld sonar within a submarine sounds is a stupid idea)
f) having to switch the batteries is annoying and it'd be much better to be able to use the nav terminal for that (but it should be disabled by default, except for vanilla subs without a sonar terminal)~~ More than implemented.

9. A parameter for **reactors** that removes the automatic mode switch (only allowing manual control) to give engineers a demanding job if the sub's creator wishes to do so.

10. A "**Power Flow Component**" that limits max (and maybe min) power that flows through it. (credits: Shiro)

11. **40mm** grenades should be fabricated **x2** because they deal less than half the damage of UEX and grenades.

12. Machines should use **less than full power** when not operating at full capacities, such as **Oxygen Generators** (power proportional to used oxygen), Charging Ports, wired Oxygen Tank Shelves/Diving Suit Lockers. The last 3 should use no power when not charging items. Oxygen Generators should be reworked to use much less power depending on oxygen levels, oxygen usage in specific rooms and whether they're charging oxygen tanks.

**Comment by Udrakan** [comment]
what makes those "best" solutions?

**Reply by Krzeszny** [reply]
Ha, I was expecting someone to question the bestness. They're my best suggestions. Simple as that.

**Comment by diegoyocys** [comment]
1. you can simply disconnect the power in cable probably. but even if you do, it might/certainly cause over voltage in the previous junction box connected to it
2. they set this game wiring, in particular, very realistic (or at least i think), sooo many electrical components, unplugging a cable from one end, doesnt mean that you will be able to pull it out, you need to unplug the other end too...., to trace it properly, you need to read the connection name, such (pump), then head down to the pump you think it is, and check if its got an unlabeled cable, since you already disconnected the other end...
3. probing wire signals? if assuming it means to test the electrical conductivity of a cable, why would you need it? 
6. its very specific, and besides, from 5% to 0% goes in like 15-20 secs?, if you choose any less, you might get caught off guard, with an inventory full (so you cant swap the tank, assuming you have no other empty tanks, or half used), being attacked, or out of oxygen, so a friendly needs to get close to provide it to you.
8. the sub sonar does detect minerals to some degree, and much further than hand-held, you just need to know how to read the colours and intensity of it.
9. do you mean the already existing switch that you can flip from automatic to manual?

**Reply by Krzeszny** [reply]
> 1. you can simply disconnect the power in cable probably. but even if you do, it might/certainly cause over voltage in the previous junction box connected to it
> 2. they set this game wiring, in particular, very realistic (or at least i think), sooo many electrical components, unplugging a cable from one end, doesnt mean that you will be able to pull it out, you need to unplug the other end too...., to trace it properly, you need to read the connection name, such (pump), then head down to the pump you think it is, and check if its got an unlabeled cable, since you already disconnected the other end...
> 3. probing wire signals? if assuming it means to test the electrical conductivity of a cable, why would you need it?
> 4. its very specific, and besides, from 5% to 0% goes in like 15-20 secs?, if you choose any less, you might get caught off guard, with an inventory full (so you cant swap the tank, assuming you have no other empty tanks, or half used), being attacked, or out of oxygen, so a friendly needs to get close to provide it to you.
> 5. the sub sonar does detect minerals to some degree, and much further than hand-held, you just need to know how to read the colours and intensity of it.
> 6. do you mean the already existing switch that you can flip from automatic to manual?

1. While I agree, one of my points was to make the repair UI less boring.
2. If I didn't make that clear enough (I'll edit it), I did not suggest being able to take wires without unplugging both ends. I wanted to be able to reconnect one end without disconnecting both ends like you can do if you double-click a disconnected end in the editor.
3. I don't think you've ever used components in Barotrauma.
4. I'm OK with no oxygen low beep at all, actually, and I'd prefer no beep than listening to it for X seconds. The thing is that those 15-20 seconds are literally _useless_. An alarm beep is supposed to get you to do something the moment you hear it. With the oxygen low beep, you're supposed to replace the oxygen tank after 15-20 seconds from hearing the continuous beep, or else the non-empty tank won't stack.
5. I didn't know that.
6. No, the lever.

**Comment by Hinny** [comment]
I think these are excellent suggestions. Devs, implement please :)

**Comment by Krzeszny** [comment]
2.5 years later and 4 of these have been implemented or the underlying issue has been solved. Not bad, not great.
